src/main/kotlin/com/hibob/rest_API/petController.kt

```diff
@@ -0,0 +1,79 @@
+package com.hibob.rest_API
+
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import java.util.*
+
+val pets = mutableListOf(
+    Pet(1L, "Buddy", "Dog", "101", Date()),
+    Pet(2L, "Whiskers", "Cat", "102", Date()),
+    Pet(3L, "Polly", "Parrot", "103", Date())
+)
+
+
+
+@Controller
+@Path("/api/matan/pets/envelopes")
+@Produces(MediaType.APPLICATION_JSON)
+class PetController {
+
+    @GET
+    @Path("/getPetType/{petId}")
```

Comment:
In REST there is no need to say what the route does so explicitly. So if it's a get already there is no need really to have it as part of the path.

Also the scopes of REST start from the biggest to the smallest detail.
So if the main route is  `/api/matan/pets` then you would then do `/{petId}` and then add later what you want from the specific pet. 
So this `Path` is more correct if it's `/{petId}/type` or `/{petId}/petType`.